src/pyfmi/fmi_util.pyx

```diff
@@ -1000,99 +1000,81 @@ cdef class DumpData:
             if self.bool_size > 0:
                 b = self.model.get_boolean(self.bool_var_ref).astype(float)
                 self.dump_data(b)
-            
-"""
-cdef class FastPointSave:
-    cdef np.ndarray real_var_ref, 
-    cdef np.ndarray int_var_ref, 
-    cdef np.ndarray bool_var_ref
-    cdef np.ndarray data_order
-    cdef np.ndarray _cache
-    cdef int nvariables
-    cdef object model
-    cdef list cache
-    cdef np.ndarray data
-    cdef int rs, ris, ribs, used_length_cache
+                
+
+from libc.stdio cimport *                                                                
+
+cdef extern from "stdio.h":
+    FILE *fdopen(int, const char *)
+
+DTYPE = np.double
+ctypedef np.double_t DTYPE_t
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def read_trajectory(file_name, int data_index, int file_position, int sizeof_type, int nbr_points, int nbr_variables):
+    cdef int i = 0
```

Comment:
I miss a docstring describing the function and the arguments.

Reply:
Fixed.